### PR TITLE
Handle socket timeout

### LINKED ISSUE #20: Doesn't retry on socket.timeout exceptions: just fails completely
Due to [this](https://github.com/kennethreitz/requests/issues/1236) requests issue and is not going to be fixed.

### PR DESCRIPTION
**Pull request by StevenMaude**
Fixes #20; not handled in `requests`.

### CODE REVIEW

File: dshelpers.py [hunk 1]
```diff
@@ -159,7 +160,7 @@ def _download_with_backoff(url, **kwargs):
     for n in range(0, _MAX_RETRIES):
```
Comment (scraperdragon) [inline dshelpers.py]: consider xrange; _MAX_RETRIES might be set to a stupid number, e.g. 99999999

Comment (StevenMaude) [inline dshelpers.py]: We could make another pull request to fix this.
